src/server/page_server.cpp

```diff
@@ -557,17 +567,17 @@ page_server::disconnect_tran_server_async (const connection_handler *conn)
 	      m_async_disconnect_handler.disconnect (std::move (*it));
 	      assert (*it == nullptr);
 	      m_passive_tran_server_conn.erase (it);
-	      passive_tran_server_found = true;
 	      break;
 	    }
 	}
-      assert (passive_tran_server_found);
```

Comment:
Now, it's possible. If the connection is not in the container, it means other thread has also done it.